pkg/reconciler/workload/placement/placement_controller.go

```diff
@@ -269,31 +269,16 @@ func (c *controller) enqueueSyncTarget(obj interface{}) {
 		return
 	}
 
-	cluster, err := c.logicalClusterLister.Cluster(clusterName).Get(corev1alpha1.LogicalClusterName)
-	if err != nil {
-		runtime.HandleError(err)
-		return
-	}
-
-	// placements referencing by cluster name
-	placements, err := c.placementIndexer.ByIndex(bySelectedLocationPath, logicalcluster.From(cluster).String())
+	// Get all locations in the same cluster and enqueue locations.
+	locations, err := c.locationLister.Cluster(clusterName).List(labels.Everything())
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
-	if path := cluster.Annotations[core.LogicalClusterPathAnnotationKey]; path != "" {
-		// placements referencing by path
-		placementsByPath, err := c.placementIndexer.ByIndex(bySelectedLocationPath, path)
-		if err != nil {
-			runtime.HandleError(err)
-			return
-		}
-		placements = append(placements, placementsByPath...)
-	}
 
 	logger := logging.WithObject(logging.WithReconciler(klog.Background(), ControllerName), obj.(*workloadv1alpha1.SyncTarget))
-	for _, placement := range placements {
-		c.enqueuePlacement(placement, logger, " because of SyncTarget")
+	for _, location := range locations {
+		c.enqueueLocation(location, logger, " because of SyncTarget")
```

Comment:
Instead of managing the source (==reason) of the `enqueue` by a suffix, can't we simply add **distinct** values ?
Something like: `logger.WithValues(logging.FromPrefix("locationReason"))` and `logger.WithValues(logging.FromPrefix("syncTargetReason"))` ?

So you can accumulate the reason objects if needed, and would have always a fixed log message.